src/layouts/slab.py

```diff
@@ -53,3 +51,68 @@
 SLAB_LAYOUT = cStruct("header" / SLAB_HEADER_LAYOUT, "nodes" / SLAB_NODE_LAYOUT[lambda this: this.header.bump_index])
 
 ORDER_BOOK_LAYOUT = cStruct(Padding(5), "account_flags" / ACCOUNT_FLAGS_LAYOUT, "slab_layout" / SLAB_LAYOUT, Padding(7))
+
+
+class Slab:
+    def __init__(self, header, nodes):
+        self._header = header
+        self._nodes = nodes
+
+    @staticmethod
+    def decode(buffer: bytes):
+        slab_layout = SLAB_LAYOUT.parse(buffer)
+        return Slab(slab_layout.header, slab_layout.nodes)
+
+    def get(self, key: int):
+        if self._header.leaf_count == 0:
+            return None
+        index: int = self._header.root
+        while True:
+            # This contains `tag` and `node`.
+            slab_node = self._nodes[index]
+            node_type: int = slab_node.tag
+            node = slab_node.node
+            # None of tree node or inner node
+            if node_type not in (1, 2):
+                raise Exception("Cannot find " + str(key) + " in slab.")
+
+            # Node key is in bytes, convert it to int.
+            node_key: int = int.from_bytes(node.key, "little")
+            # Leaf Node that matches
+            if node_type == 2:  # pylint: disable=no-else-return
+                if node_key == key:
+                    return node
+                return None
+            elif node_type == 1:
+                if (node_key ^ key) >> (128 - slab_node.node.prefix_len) != 0:
+                    return None
+                # Check if the n-th bit (start from the least significant, i.e. rightmost) of the key is set
+                index = node.children[(key >> (128 - node.prefix_len - 1)) & 1]
```

Comment:
Raise exception for bad node_type.

Reply:
I think the tuple check is to make sure that we can access the `node.key`, if we moved the Exception part to the `else` branch, we will need to convert the `key` to `node_key` twice. in both branches for `leaf` and `inner_node`.

Reply:
Could you raise a runtime exception stating that execution should not go beyond the last conditional?

Reply:
Added `RunTimeError`.